packages/material-ui/src/StepButton/StepButton.test.js

```diff
@@ -1,122 +1,131 @@
-            onMouseEnter={handleMouseEnter}
-            onMouseLeave={handleMouseLeave}
-            {...defaultProps}
-          >
-            Step One
-          </StepButton>,
-        );
-        wrapper.simulate('mouseEnter');
-        assert.strictEqual(handleMouseEnter.callCount, 1);
-        wrapper.simulate('mouseLeave');
-        assert.strictEqual(handleMouseEnter.callCount, 1);
-        assert.strictEqual(handleMouseLeave.callCount, 1);
-      });
+      const stepLabel = wrapper.find(StepLabel);
+      expect(stepLabel.props()).to.have.property('active', true);
+      expect(stepLabel.props()).to.have.property('completed', true);
+      expect(stepLabel.props()).to.have.property('disabled', true);
+      expect(stepLabel.props()).to.have.property('children', 'Step One');
     });
 
-    it('should bubble callbacks used internally', () => {
+    it('should pass props to a provided StepLabel', () => {
+      const wrapper = mount(
+        <StepButton active completed disabled label="Step One">
+          <StepLabel>Step One</StepLabel>
+        </StepButton>,
+      );
+
+      const stepLabel = wrapper.find(StepLabel);
+      expect(stepLabel.props()).to.have.property('active', true);
+      expect(stepLabel.props()).to.have.property('completed', true);
+      expect(stepLabel.props()).to.have.property('disabled', true);
+      expect(stepLabel.props()).to.have.property('children', 'Step One');
+    });
+  });
+
+  it('should disable the button', () => {
+    const { getByRole } = render(<StepButton disabled>Step One</StepButton>);
+
+    expect(getByRole('button')).to.have.property('disabled', true);
+  });
+
+  describe('event handlers', () => {
+    it('should forward mouseenter, mouseleave and touchstart', () => {
       const handleMouseEnter = spy();
       const handleMouseLeave = spy();
       const handleTouchStart = spy();
-      const wrapper = shallow(
+      const { getByRole } = render(
         <StepButton
           onMouseEnter={handleMouseEnter}
           onMouseLeave={handleMouseLeave}
           onTouchStart={handleTouchStart}
-          {...defaultProps}
         >
           Step One
         </StepButton>,
       );
-      wrapper.simulate('mouseEnter');
-      assert.strictEqual(handleMouseEnter.callCount, 1);
-      wrapper.simulate('mouseLeave');
-      assert.strictEqual(handleMouseEnter.callCount, 1);
-      assert.strictEqual(handleMouseLeave.callCount, 1);
-      wrapper.simulate('touchStart');
-      assert.strictEqual(handleMouseEnter.callCount, 1);
-      assert.strictEqual(handleMouseLeave.callCount, 1);
-      assert.strictEqual(handleTouchStart.callCount, 1);
-      wrapper.simulate('mouseEnter');
-      wrapper.simulate('touchStart');
-      assert.strictEqual(handleMouseEnter.callCount, 2);
-      assert.strictEqual(handleMouseLeave.callCount, 1);
-      assert.strictEqual(handleTouchStart.callCount, 2);
+      const button = getByRole('button');
+
+      fireEvent.mouseOver(button);
```

Comment:
Something about mouseenter vs mouseover. Need a refresher what the difference is and how react handles those.